source/configuration_manual/event_filter.rst

```diff
@@ -0,0 +1,119 @@
+.. _event_filter:
+
+===============
+Event Filtering
+===============
+
+Dovecot's event support includes the ability to narrow down which events are
+processed by filtering them based on the administrator-supplied predicate.
+
+Individual events can be identified either by their name or source code
+location.  The source location of course can change between Dovecot
+versions, so it should be avoided.
+
+Regardless of the syntax used, matching is performed the same way:
+
+* Event names are compared using a case-sensitive wildcard match.  The
+  wildcards supported are ``?`` and ``*``.
+* Event location is compared in two parts: the file name is compared
+  case-sensitively, and the line number is compared as an integer.  For a
+  match to occur, the filename match *and* the line number must either match
+  or be unspecified.
+* Event categories are compared using a "has a" relationship.  A category in
+  the filter must be present in the event for a match to occur.  Any other
+  categories on the event do not influence the match.
+* Event fields are compared using a case-insensitive wildcard match.  The
+  wildcards supported are ``?`` and ``*``.
+
+.. _event_filter_metric:
+
+Metric filter syntax
+^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: v2.3
+
+Events can be filtered inside the ``metric`` blocks (see :ref:`statistics`)
+based on the event name, source location, the categories present, and field
+values.
+
+All four use the same ``key=value`` syntax, however the semantics of each
+are slightly different.
+
+* Event name filtering uses the ``event_name`` key.  The value is matched as
+  described above.
+* Event source location filtering uses the ``source_location`` key.  The
+  value is matched as described above.
+* Event category filtering uses the ``categories`` key.  The value is a
+  space-separated list of categories *all* of which must be matched as
+  described above.
+* Event field filtering uses the field name as the key, however the
+  key-value pairs are inside the ``filter`` sub-block.  The value is matched
+  as described above.
+
+An event is said to match the filter if *all* of the specified key-value
+pairs match.
+
+For example, the following matches all events with the name
+``abc``, the source code location ``def.c:123``, the categories ``ghi`` and
+``jkl``, and the field ``mno`` equal to 456 and ``pqr`` beginning with the
+string ``xyz123*``::
+
+   metric foo {
+     event_name = abc
+     source_location = def.c:123
+     categories = ghi jkl
+     filter {
+       mno = 456
+       pqr = xyz123*
```

Comment:
How about a more sensible example, e.g.:

```
metric example_http_metric {
  event_name = http_request_finished
  source_location = http-client.c:123
  categories = storage imap
  filter {
    status_code = 200
    attempts = 1
```

Reply:
Oh, that example doesn't use a wildcard. Maybe replace attempts with .. lets say "user = testuser*"

Reply:
I purposefully used a made up event to avoid having this example get out of sync with the real event.  Since I don't care enough about that, I changed it up to your example.